### PR TITLE
README.md: add compatibility matrix

### PR DESCRIPTION
**Pull request by runcom**
@mrunalp @sameo PTAL

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by jawnsy** [comment]
@runcom Nice! How about using words like "Active" and "Security-only" instead of symbols? Does "maintenance is manual" mean "backports/maintenance only on request" or does it mean "we will evaluate all patches but only backport severe security ones"?

Also doesn't it go without saying that `master` is the active development branch, and therefore gets everything?

File: README.md [hunk 1]
```diff
@@ -42,6 +42,19 @@ For async communication and long running discussions please use issues and pull
 
 For sync communication we have an IRC channel #cri-o, on chat.freenode.net, that everyone is welcome to join and chat about development.
 
+## Compatibility matrix: CRI-O <-> Kubernetes clusters
+
+| Version - Branch        | Kubernetes branch/version     | Maintenance status |
+|-------------------------|-------------------------------|--------------------|
+| CRI-O 1.0  - kube-1.6.x | Kubernetes 1.6 branch, v1.6.x | =                  |
```
Comment (sameo) [inline README.md]: `1.0.x` ?

File: README.md [hunk 2]
```diff
@@ -42,6 +42,19 @@ For async communication and long running discussions please use issues and pull
...
+| CRI-O 1.1  - kube-1.7.x | Kubernetes 1.7 branch, v1.7.x | ✓                  |
```
Comment (sameo) [inline README.md]: `1.1.x` ?

File: README.md [hunk 3]
```diff
@@ -42,6 +42,19 @@ For async communication and long running discussions please use issues and pull
...
+| CRI-O HEAD - master     | Kubernetes master branch      | ✓                  |
+
+Key:
+
+* `✓` Changes in main Kubernetes repo about CRI are actively implemented in CRI-O
```
Comment (sameo) [inline README.md]: The CRI changes being tracked and implemented is the most important part, but we obviously also are going to keep the overall compatibility with Kubernetes. May be this should be explicitly called out.

File: README.md (hunk 2 above)
Comment (sameo) [inline README.md]: Should we add this line to the matrix once the branch is created ?

File: README.md (hunk 1 above)
Comment (sameo) [inline README.md]: So this looks like as if 1.0 is already released and the `kube-1.6.x branch` is already in maintenance mode.

**Comment by sameo** [comment]
One crazy proposition: After 1.0.0, should we consider tagging CRI-O with 1.N.x, where N tracks the Kubernetes release it is compatible to ? So the next release after 1.0.0 would be 1.7.0. It would make it even easier for our users to track which Kubernetes version a given CRI-O release is compatible with.

File: README.md (hunk 1 above)
Comment (runcom) [inline README.md]: right, it isn't. We can leave this PR in flight till we release though.

File: README.md (hunk 2 above)
Comment (runcom) [inline README.md]: yup, let's leave this PR open till 1.0.0 actually.

File: README.md (hunk 3 above)
Comment (runcom) [inline README.md]: sounds good

**Comment by runcom** [comment]
> One crazy proposition: After 1.0.0, should we consider tagging CRI-O with 1.N.x, where N tracks the Kubernetes release it is compatible to ? So the next release after 1.0.0 would be 1.7.0. It would make it even easier for our users to track which Kubernetes version a given CRI-O release is compatible with.

I've been thinking about this for weeks now, and I'm strongly in favor of this after the first 1.0.0 release

**Comment by mrunalp** [comment]
Sounds great! 👍

**Comment by runcom** [comment]
@sameo it was that crazy proposition at the end :)

**Comment by rhatdan** [comment]
I totally agree that releases of CRI-O versions should exactly match the kubernetes versions they are for, this eliminates us from having to deal with a compatability matrix.

**Comment by runcom** [comment]
@mrunalp @rhatdan updated this, please merge

**Comment by sameo** [comment]
LGTM

**Comment by rhatdan** [comment]
LGTM
Doc change, no reason to run tests.

**Pull request merged**